### ISSUE TITLE
HTTP Query response object delayed loading

### ISSUE THREAD

**Issue by serbanman** [ToolJet/ToolJet#11192]
### Version Information
3.0.0beta

### Environment
ToolJet Docker

### What is the expected behaviour?
call
`
await query.someQuery.run()
let responseStatusCode = query.someQuery.response.statusCode
`
and obtain proper response status code

### What is the current behaviour?
query.someQuery.response is undefined until JS exits

### How to reproduce the issue?

1. make a http query with parameter 'expectedCode':
https://httpbin.org/status/{{parameters.expectedCode}}
3. make a JS script:
`
await queries.testHTTPCall.run({expectedCode: 401})
alert(queries.testHTTPCall.response?.statusCode)

await queries.testHTTPCall.run({expectedCode: 402})
alert(queries.testHTTPCall.response?.statusCode)

await queries.testHTTPCall.run({expectedCode: 403})
alert(queries.testHTTPCall.response?.statusCode)
`
one may expect to see alerts with 401, 402, 403, but in fact it is undefined, undefined, undefined on the first run, and 403, 403, 403 on the second and so on. 

With this approach there is no way to do any sort of conditional expressions based on HTTP status codes, because they are never relevant.

### Any possible solutions?
Maybe add awaitable .getResponse() to the query object?

**Comment by SarvT** [comment]
Yes, maybe awaitable .getResponse() would work or it will need a custom wrapper function. Can this issue be assigned to me? I  would like to work on it.

**Comment by PriteshKiri** [comment]
Hey @serbanman,

Thanks for creating the issue.

We will look into it and will get back to you soon.
